### PR TITLE
refactor PathItemIcon/PathItemInfo into their own connected components

### PR DESCRIPTION
**Pull request by songgao**

### CODE REVIEW

**Comment by songgao** [comment]
untagging review as I'm gonna move badges into the component too

**Comment by songgao** [comment]
ready again!

File: shared/fs/row/editing-container.js [hunk 1]
```diff
@@ -11,14 +11,9 @@ type OwnProps = {
...
-const mapStateToProps = (state, {editID}: OwnProps) => {
-  const _edit = state.fs.edits.get(editID, Constants.makeNewFolder()) // TODO make missing get better
-  const _username = state.config.username
-  return {
-    _edit,
-    _username,
-  }
-}
+const mapStateToProps = (state, {editID}: OwnProps) => ({
+  _edit: state.fs.edits.get(editID, Constants.makeNewFolder()),
```
Comment (chrisnojima) [inline shared/fs/row/editing-container.js]: should move makeNewFolder into a const above so it doens't thrash

File: shared/fs/common/path-item-icon.js [hunk 2]
```diff
@@ -2,67 +2,140 @@
...
+  <Kb.Icon type="iconfont-folder-private" color={Styles.globalColors.grey} fontSize={props.size} />
+)
+
+// $FlowIssue
+const i = (s: string): Kb.IconType => s
+
+const IconOnly = (props: Props) => {
+  const elems = Types.getPathElements(props.path)
+
+  if (elems.length === 1) {
```
[the remark this answers is not in the document]
Comment (songgao) [reply shared/fs/common/path-item-icon.js]: I wonder if we should just replace `FsTypes.Path` (which is currently an opaque string) with an immutable data structure that looks like what you listed above. Is there a reason not to do that?

**Comment by songgao** [comment]
@chrisnojima ready for another look!

**Comment by songgao** [comment]
Cool thanks!

**Pull request merged**